src/content/concepts/under-the-hood.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Under The Hood
+sort: 1
+contributors:
+  - smelukov
+---
+
+> This section describes webpack internals and useful for plugin developers
+
+The bundling is a function that takes some files and emits others.
+
+But between input and output, it also has [modules](/concepts/modules/), [entry points](/concepts/entry-points/), chunks, chunk groups, and many other intermediate parts.
+
+## The main parts
+
+Every file used in your project is a [Module](/concepts/modules/)
+
+__./index.js__
+
+```js
+import app from './app.js';
+```
+
+__./app.js__
+
+```js
+export default 'the app';
+```
+
+By using each other, the modules form a graph (`ModuleGraph`). 
```

Comment:
Is there any docs to explain what a ModuleGraph is?

Reply:
Not yet. In progress